src/components/Dropdown.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import './Dropdown.css';
+
+class Dropdown extends React.Component {
+  componentWillMount = () => document.addEventListener('click', this.handleClick, false);
```

Comment:
you can remove the listener on willMount and willUnmount if we are using the ones on willReceiveFront